linera-execution/src/runtime.rs

```diff
@@ -1032,6 +1036,26 @@ impl<UserInstance> BaseRuntime for SyncRuntimeInternal<UserInstance> {
             .push(OracleResponse::Blob(*blob_id));
         Ok(blob_content)
     }
+
+    fn contains_blob(&mut self, blob_id: &BlobId) -> Result<bool, ExecutionError> {
+        if let Some(responses) = &mut self.replaying_oracle_responses {
+            match responses.next() {
+                Some(OracleResponse::Blob(oracle_blob_id)) if oracle_blob_id == *blob_id => {}
+                Some(_) => return Err(ExecutionError::OracleResponseMismatch),
+                None => return Err(ExecutionError::MissingOracleResponse),
+            }
+        }
+        let exists = self
+            .execution_state_sender
+            .send_request(|callback| ExecutionRequest::ContainsBlob {
+                blob_id: *blob_id,
+                callback,
+            })?
+            .recv_response()?;
```

Comment:
This could return different results in different nodes _even if_ there are recorded oracle responses.

I'd rather make `contains_blob` behave exactly like `read_blob`, and fail if it can't be found. So maybe the system API and runtime functions should be `assert_blob`?